python-threatexchange/threatexchange/signal_type/text_md5.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Wrapper around the Text MD5 signal types.
+"""
+
+import typing as t
+import hashlib
+import pathlib
+from threatexchange.content_type.content_base import ContentType
+from threatexchange.content_type.text import TextContent
+
+from threatexchange.signal_type import signal_base
+from threatexchange import common
+from threatexchange.signal_type.raw_text import RawTextSignal
+from threatexchange.fetcher.apis.fb_threatexchange_signal import (
+    HasFbThreatExchangeIndicatorType,
+)
+
+
+class TextMD5Signal(
+    signal_base.SimpleSignalType,
+    signal_base.TextHasher,
+    signal_base.FileHasher,
+    HasFbThreatExchangeIndicatorType,
+):
+    """
+    Simple signal type for Text MD5s.
+    """
+
+    INDICATOR_TYPE = "TEXT_URL_MD5"
+
+    @classmethod
+    def get_content_types(self) -> t.List[t.Type[ContentType]]:
+        return [TextContent]
+
+    @classmethod
+    def hash_from_str(cls, text: str) -> str:
+        encoded_text = common.normalize_string(text).encode("utf-8")
+        text_hash = hashlib.md5(encoded_text)
+        return text_hash.hexdigest()
+
+    @classmethod
+    def hash_from_file(cls, path: pathlib.Path) -> str:
```

Comment:
nit: `hash text` from file is pretty confusing for md5. As a user I wouldn't be sure without checking the code if this option would give me: an MD5 of the file, an MD5 of each line, or an MD5 of the string of text in the file.

cc: @Dcallies how did you image this command being used?

Reply:
I believe the MD5 of text is the same as MD5 of file. Unless there are encoding weirdnesses.
<img width="361" alt="Screen Shot 2022-04-07 at 11 22 04" src="https://user-images.githubusercontent.com/217056/162234684-c67c10a6-858e-48b6-bede-65c2cdd52117.png">

Reply:
Thanks for clarifying! 
White space might have a view gotcha but this address my main concern